GDJS/Runtime/runtimegame.ts

```diff
@@ -520,73 +523,77 @@ namespace gdjs {
-          // Most browser/engines will try to run at slightly more than 60 frames per second.
-          // If game is set to have a maximum FPS to 60, then one out of two frames will be dropped.
-          // Hence, we use a 7 frames margin to ensure that we're not skipping frames too much.
-          return true;
-        }
-        const elapsedTime = accumulatedElapsedTime;
-        accumulatedElapsedTime = 0;
+        //Uncomment to profile the first x frames of the game.
+        // var x = 500;
+        // var startTime = Date.now();
+        // console.profile("Stepping for " + x + " frames")
+        // for(var i = 0; i < x; ++i) {
+        //     this._sceneStack.step(16);
+        // }
+        // console.profileEnd();
+        // var time = Date.now() - startTime;
+        // console.log("Took", time, "ms");
+        // return;
+
+        //The standard game loop
+        const that = this;
+        let accumulatedElapsedTime = 0;
+        this._renderer.startGameLoop(function (lastCallElapsedTime) {
+          if (that._paused) {
+            return true;
+          }
 
-        //Manage resize events.
-        if (that._notifyScenesForGameResolutionResize) {
-          that._sceneStack.onGameResolutionResized();
-          that._notifyScenesForGameResolutionResize = false;
-        }
+          // Skip the frame if we rendering frames too fast
+          accumulatedElapsedTime += lastCallElapsedTime;
+          if (
+            that._maxFPS > 0 &&
+            1000.0 / accumulatedElapsedTime > that._maxFPS + 7
+          ) {
+            // Only skip frame if the framerate is 7 frames above the maximum framerate.
+            // Most browser/engines will try to run at slightly more than 60 frames per second.
+            // If game is set to have a maximum FPS to 60, then one out of two frames will be dropped.
+            // Hence, we use a 7 frames margin to ensure that we're not skipping frames too much.
+            return true;
+          }
+          const elapsedTime = accumulatedElapsedTime;
+          accumulatedElapsedTime = 0;
 
-        //Render and step the scene.
-        if (that._sceneStack.step(elapsedTime)) {
-          that.getInputManager().onFrameEnded();
-          return true;
-        }
-        return false;
-      });
-      setTimeout(() => {
-        this._setupSessionMetrics();
-      }, 10000);
+          //Manage resize events.
+          if (that._notifyScenesForGameResolutionResize) {
+            that._sceneStack.onGameResolutionResized();
+            that._notifyScenesForGameResolutionResize = false;
+          }
+
+          //Render and step the scene.
+          if (that._sceneStack.step(elapsedTime)) {
+            that.getInputManager().onFrameEnded();
+            return true;
+          }
+          return false;
+        });
+        setTimeout(() => {
+          this._setupSessionMetrics();
+        }, 10000);
+      } catch (e) {
```

Comment:
Don't put a giant try/catch here, we want crash to happen without being swallowed or risking calling a function of the engine that would itself trigger another crash. Basically, if a crash happen here, everything is already lost. At the very least, rethrow the error because it should surface to the traditional JS console and debugger.

Reply:
Well in case of a crash it is easier for the user to use the integrated tooling to send us the error message rather than having them to open the js debugger etc so I would keep the try catch

Reply:
What do you think of at least `throw`ing again the error?